### PR TITLE
feat: escape the avatal URL in CAS response

### PR DESCRIPTION
**Pull request by nanoguo**
Fix: https://github.com/casdoor/casdoor/issues/3411

Is it ok if I only escape the avatar url?

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

File: object/token_cas.go [hunk 1]
```diff
@@ -225,6 +235,11 @@ func GenerateCasToken(userId string, service string) (string, error) {
 		}
 
 		if value != "" {
+			if k == "avatar" {
```
Comment (dacongda) [inline object/token_cas.go:238]: all value will be string, maybe escape all value could be a good idea

Comment (nanoguo) [reply object/token_cas.go:238]: It's ok to escape all value. if other values have special characters such as '&', they can also make crash.

**Comment by hsluoyz** [comment]
@dacongda

File: object/token_cas.go [hunk 2]
```diff
@@ -225,6 +235,9 @@ func GenerateCasToken(userId string, service string) (string, error) {
 		}
 
 		if value != "" {
+			if escapedValue, err := escapeXMLText(value); err == nil {
```
Comment (hsluoyz) [inline object/token_cas.go:238]: What error will it generate? We usually return error if error found

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/token_cas.go (hunk 2 above)
Comment (nanoguo) [reply object/token_cas.go:238]: The error only occur in io.Write function, this escapeXMLText function is try to escape a original string to xml safe string, if it failed, it did nothing, and the response will use the original string like before. should I catch the error and break loop?

Comment (hsluoyz) [reply object/token_cas.go:238]: @nanoguo if that error is unpected and critical, we should handle it, not ignore it

Comment (nanoguo) [reply object/token_cas.go]: ok

**Comment by hsluoyz** [comment]
@dacongda

**Review by dacongda — APPROVED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**